src/main/java/com/rcv/GuiConfigController.java

```diff
@@ -243,7 +245,7 @@ public void buttonTabulateClicked() {
     if (checkForSaveAndTabulate()) {
       if (GuiContext.getInstance().getConfig() != null) {
         buttonBar.setDisable(true);
-        TabulatorService service = new TabulatorService();
+        TabulatorService service = new TabulatorService(selectedFile.getAbsolutePath());
```

Comment:
I believe this is correct encapsulation.  TabulatorService needs to know about TabulationSession and how to construct one.  It does not need to know about GUIContext.  Why add that dependency?

Reply:
It just seemed weird to me that we create a `TabulatorSession`object in line 178, but then we create a new one when we actually execute the tabulation. Maybe pass in the existing `TabulatorSession` instead of the path when initializing `TabulatorService`? Or maybe it's not a big deal? Just seemed like code smell to me.

Reply:
Gotcha.  Yes, your instinct is correct.  The code needs another piece of cleanup to refactor how ContestConfig objects are created and validated, so they can be used aside from TabulatorSession.  When that's done, the TabulatorSession reference in line 178 will be replaced with something like:
ContestConfig config = ContestConfig.loadContestConfig(fileToLoad.getAbsolutePath);